arch/risc-v/src/mpfs/mpfs_head.S

```diff
@@ -174,6 +174,10 @@ __start_mpfs:
 
 .continue_boot:
 
+    /* Clear PMP */
+
+    csrw pmpcfg0, zero
+    csrw pmpcfg2, zero
```

Comment:
why these registers and why here?

Reply:
I mean, could all the pmp configs be done in riscv_pmp, and called somewhere from mpfs startup code later on, depending on related config flags.

Reply:
The registers are what hold all pmp configurations, wiping them will reset pmp.

The problem is that after a warm reset the pmp registers still hold any old configurations. I put these here to simply guarantee that the system starts from a known state each time.

One option is of course to setup pmp to a known configuration later on. However, the way it is done now (1. test if configuration exists, 2. do configuration if not) does not quite work for this purpose after warm resets (because the configuration already exists). This is not an issue if the configuration is not modified, but becomes one if it is modified (which happens e.g. when the memory map is changed).